File: Programmers/42576.py
# 완주하지 못한 선수


def solution(participant, completion):
    race = {}
    for name in participant:
        if name in race:
            race[name] += 1
        else:
            race[name] = 1

    for name in completion:
        race[name] -= 1

    for key, val in race.items():
        if val != 0:
            return key

File: Programmers/42576.js
// 완주하지 못한 선수 - 해시
function solution(participant, completion) {
    let person = {}
    for (let p of participant) {
        p in person ? person[p]++ : person[p] = 1
    }
    for (let p of completion) {
        person[p]--;
    }
    for (let p in person) {
        if (person[p] > 0)
            return p;
    }
}